Secrets Import documentation (#25594)
hashicorp/vault: website/content/docs/import/gcpsm.mdx

---
layout: docs
page_title: Google Cloud Platform Secret Manager - Secrets Import Source
description: The Google Cloud Platform Secret Manager source imports secrets from GCP to Vault.

> ---

Reviewer: The general documentation typically uses a walkthrough style that guides the user through a full-working example from start to finish. Do you think the same pattern would be useful for Secrets Import?

Author: I had first copied our usual format based on other docs to base these on, but I started to notice when writing the GCP page that other source pages would be nearly identical. The only section that would be different is explaining the source block with the rest being unchanged. I figure it makes sense to have a small page for each source block, a page for the mappings, and put the destination info on the index page since we are only dealing with Vault as a destination.

# GCP import source

The GCP source enables users to import secret data from GCP Secret Manager into their Vault instance. To be able
to import secret data from here into your Vault destination, the following permissions are required:

```shell-session
"secretmanager.secrets.list",
"secretmanager.versions.access",

> ```

Reviewer: In the Secrets Sync doc this was put into a `Permissions` section, should we do the same?

## Argument reference

* `name` - (Required) The name of the source block used to internally reference this source inside of a mapping.

* `credentials` - (Optional) The path to the service account key credentials file for the service account with the ncessary
permissions. If `credentials` is set, then `vault_mount_path` and `vault_role_name` must be unset.

* `vault_mount_path` - (Optional) The Vault mount path to a preconfigured GCP secrets engine that is used to generate a
dynamic credential for the importer to use. If `vault_mount_path` or `vault_role_name` are set, then `credentials` must be unset.

* `vault_role_name` - (Optional) The Vault role's name that exists in the preconfigured GCP secrets engine mount that is
used to generate a dynamic credential for the importer to use. If `vault_role_name` or `vault_mount_path` are set, then
`credentials` must be unset.

Refer to a [specific import source's documentation](#sources) for details on how to define all remaining arguments of a source

> block for that external system type.

Reviewer: I think this link does not work?

## Example

In this example, a GCP source is defined and configured to use 

```hcl
source_gcp {
  name = "my-gcp-source-1"
  secrets_engine_mount = "gcp"

>   secrets_engine_role = "my-gcp-role-1"

Reviewer: Should we link users to the GCP secrets engine or give them a simplified rundown of how to set it up?

---
layout: docs

> page_title: Google Cloud Platform Secret Manager - Secrets Import Source

Author: Great point

description: The Google Cloud Platform Secret Manager source imports secrets from GCP to Vault.
---

# GCP import source

> Use the GCP source to import secret data from GCP Secret Manager into your Vault instance. To use dynamic

Reviewer 2: ```suggestion
Use the GCP source to import secret data from GCP secret manager into your Vault instance. To use dynamic
```

Author: GCP Secret Manager is the name of the product from Google, should it stay capitalized for this?

> credentials for this source, ensure that the [GCP secrets engine](/vault/docs/secrets/gcp) has

Reviewer 2: ```suggestion
credentials with GCP import, ensure the [GCP secrets engine](/vault/docs/secrets/gcp) is
```

Style correction: write in active voice

> already been configured.

Reviewer 2: ```suggestion
already configured.
```

## Argument reference

Refer to the [HCL syntax](/vault/docs/import#hcl-syntax-1) for arguments common to all source types.

## Additional arguments

- `credentials` `(string: "")` - The path to the service account key credentials file for the service account
  with the [necessary permissions](#permissions). If `credentials` is set, then `vault_mount_path` and
  `vault_role_name` must be unset.

- `vault_mount_path` `(string: "")` - The Vault mount path to a pre-configured GCP
  secrets engine used to generate dynamic credentials for the importer. If
  `vault_mount_path` or `vault_role_name` are set, then `credentials` must be
  unset.

- `vault_role_name` `(string: "")` - The Vault role used to generate a dynamic
  credential for the importer. The role name must exist in the pre-configured
  GCP secrets engine mount. If `vault_role_name` or `vault_mount_path` are set,
  then `credentials` must be unset.

## Example

Define and configure the `my-gcp-source-1` GCP source:

```hcl
source_gcp {
  name                 = "my-gcp-source-1"
  secrets_engine_mount = "gcp"
  secrets_engine_role  = "my-gcp-role-1"
}
```

## Permissions

> To use this source type, you must grant the GCP identity being used to read secrets the following permissions:

Reviewer 2: ```suggestion
To use GCP import, you must grant the associated GCP identity permission to read secrets:
```
